Build fail on windows with MSVC (v13)

I've found this issue when using `xsimd `through `vcpkg`.

I've filed a [bug in vcpkg](https://github.com/microsoft/vcpkg/issues/39810) too, but i thought i should open it here too.


```
D:\a\FW\builds\ninja-release-vcpkg\vcpkg_installed\x64-windows\include\xsimd\arch\generic./xsimd_generic_details.hpp(207,22): error C2187: syntax error: 'char' was unexpected here
D:\a\FW\builds\ninja-release-vcpkg\vcpkg_installed\x64-windows\include\xsimd\arch\generic./xsimd_generic_details.hpp(207,28): error C3878: syntax error: unexpected token '=' following 'simple_declaration'
D:\a\FW\builds\ninja-release-vcpkg\vcpkg_installed\x64-windows\include\xsimd\arch\generic./xsimd_generic_details.hpp(207,28): note: error recovery skipped: '='
D:\a\FW\builds\ninja-release-vcpkg\vcpkg_installed\x64-windows\include\xsimd\arch\generic./xsimd_generic_details.hpp(211,76): error C2760: syntax error: ')' was unexpected here; expected 'expression'
D:\a\FW\builds\ninja-release-vcpkg\vcpkg_installed\x64-windows\include\xsimd\arch\generic./xsimd_generic_details.hpp(211,76): error C2760: syntax error: ')' was unexpected here; expected ';'
D:\a\FW\builds\ninja-release-vcpkg\vcpkg_installed\x64-windows\include\xsimd\arch\generic./xsimd_generic_details.hpp(211,76): error C3878: syntax error: unexpected token ')' following 'jump_statement'
D:\a\FW\builds\ninja-release-vcpkg\vcpkg_installed\x64-windows\include\xsimd\arch\generic./xsimd_generic_details.hpp(211,76): 
```

--- a/.github/workflows/cross.yml
+++ b/.github/workflows/cross.yml
@@ -5,7 +5,7 @@ concurrency:
   cancel-in-progress: true
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     name: '${{ matrix.target.arch }}, ${{ matrix.sys.compiler }} ${{ matrix.sys.version }}'
     strategy:
       matrix:
@@ -14,7 +14,7 @@ jobs:
           - { platform: 'arm',     arch: 'armv7-a', dir: 'arm-linux-gnueabihf', flags: '-mfpu=vfpv3-d16', full: 'OFF' } # no neon
           - { platform: 'aarch64', arch: 'armv8-a', dir: 'aarch64-linux-gnu', flags: '', full: 'ON' }
         sys:
-          - { compiler: 'gcc',   version: '8' }
+          - { compiler: 'gcc',   version: '9' }
           - { compiler: 'clang', version: 'latest' }
     steps:
     - name: Setup compiler

--- a/include/xsimd/arch/generic/xsimd_generic_details.hpp
+++ b/include/xsimd/arch/generic/xsimd_generic_details.hpp
@@ -204,11 +204,11 @@ namespace xsimd
             XSIMD_INLINE batch<uint32_t, A> fast_cast(batch<float, A> const& v, batch<uint32_t, A> const&, requires_arch<generic>) noexcept
             {
                 auto is_large = v >= batch<float, A>(1u << 31);
-                auto small = bitwise_cast<float>(batch_cast<int32_t>(v));
-                auto large = bitwise_cast<float>(
+                auto small_v = bitwise_cast<float>(batch_cast<int32_t>(v));
+                auto large_v = bitwise_cast<float>(
                     batch_cast<int32_t>(v - batch<float, A>(1u << 31))
                     ^ batch<int32_t, A>(1u << 31));
-                return bitwise_cast<uint32_t>(select(is_large, large, small));
+                return bitwise_cast<uint32_t>(select(is_large, large_v, small_v));
             }
         }
 

